tests/unit_tests/advanced_data_type/types_tests.py

```diff
@@ -513,3 +517,72 @@ def test_port_translate_filter_func_not_in_double():
     assert port.translate_filter(input_column, input_operation, input_values).compare(
         port_translate_filter_response
     )
+
+
+def test_stream_id():
+    print("Testing stream ID...\n")
+    known_key = 11
+    known_key_2 = 129
+    known_value = "IIS_EXT"
+    known_value_2 = "MT-ASSEMBLYLINE-U-PROD"
+    assert list(STREAM_DICT.keys())[list(STREAM_DICT.values()).index(known_value)] == known_key
+    assert list(STREAM_DICT.keys())[list(STREAM_DICT.values()).index(known_value_2)] == known_key_2
+    assert list(STREAM_DICT.keys())[list(STREAM_DICT.values()).index(known_value)] != known_key_2
+    assert list(STREAM_DICT.keys())[list(STREAM_DICT.values()).index(known_value)] != known_key_2
+    val_req: AdvancedDataTypeRequest = {
+    "values": ["IIS_EXT"]
+    }
+    key_req: AdvancedDataTypeRequest = {
+    "values": [11]
+    }
+    test_key:AdvancedDataTypeResponse = stream_id_func(key_req)
+    test_val:AdvancedDataTypeResponse = stream_id_func(val_req)
+    assert test_key["error_message"] == ""
+    assert test_val["error_message"] == ""
+    invalid_val_req: AdvancedDataTypeRequest = {
+    "values": ["fail"]
+    }
+    invalid_key_req: AdvancedDataTypeRequest = {
+    "values": [0]
+    }
+    test_invalid_key:AdvancedDataTypeResponse = stream_id_func(invalid_key_req)
+    test_invalid_val:AdvancedDataTypeResponse = stream_id_func(invalid_val_req)
+    assert test_invalid_val['error_message'] == f"'{invalid_val_req['values'][0]}' is not a valid Stream ID. Did not match a known Stream ID or name."
+    assert test_invalid_key['error_message'] == f"'{invalid_key_req['values'][0]}' is not a valid Stream ID. Did not match a known Stream ID or name."
+    print("Stream ID passed!\n")
+
+def test_ipv6_address():
```

Comment:
Could you also add tests for the compact notation of IPv6? E.g.: `2001::0370:7334` and `::ffff:abcd:1234`